fedimint-bitcoind/src/bitcoincore_rpc.rs

```diff
@@ -47,6 +48,17 @@ pub fn from_url_to_url_auth(url: &Url) -> Result<(String, Auth)> {
         }),
         if url.username().is_empty() {
             Auth::None
+        } else if url.password().is_none() {
+            if let Some(cookie_file) = rpc_cookie_file() {
+                let cookie = std::fs::read_to_string(cookie_file)?;
+                let auth = parse_cookie_string(&cookie).unwrap_or(Auth::None);
+                auth
+            } else {
+                let cookie_file_path = format!("{}/.cookie", url.path());
+                create_cookie_file(&cookie_file_path, "bitcoin", "bitcoin")?;
+                let auth = Auth::UserPass("bitcoin".to_owned(), "bitcoin".to_owned());
+                auth
```

Comment:
@VeronicaTee Is creating the .cookie file needed/appropriate? IIRC the creation should be before `bitcoind` start and then passed to it if the user needs a custom .cookie file.

IIRC also if the user wants to use both a `-rpcpassword`,`-rpcuser` and a .cookie file, it would need to use the `-rpcauth` instead which handles both user/pass and .cookie auth.